src/ParamConverter/HashidsParamConverter.php

```diff
@@ -39,53 +36,76 @@ public function supports(ParamConverter $configuration): bool
         return true;
     }
 
-    private function setHashid(Request $request, ParamConverter $configuration): void
+    private function decodeArgumentsController(Request $request, ParamConverter $configuration): void
     {
-        $hashids = $this->hashids->decode(
-            $this->getIdentifier(
-                $request,
-                array_replace(['hashid' => null], $configuration->getOptions()),
-                (string) $configuration->getName()
-            )
-        );
+        $hash = $this->getHash($request, $configuration);
+
+        if ($this->isSkippable($hash)) {
+            return;
+        }
+
+        $name = $configuration->getName();
+        $hashids = $this->hashids->decode($hash);
 
         if ($this->hasHashidDecoded($hashids)) {
-            $request->attributes->set($configuration->getName(), current($hashids));
+            $request->attributes->set($name, current($hashids));
+        }
+
+        if (!$this->autoConvert && !$this->hasHashidDecoded($hashids)) {
+            throw new \LogicException(sprintf('Unable to decode parameter "%s".', $name));
         }
     }
 
-    private function getIdentifier(Request $request, $options, string $name): string
+    private function getHash(Request $request, ParamConverter $configuration): ?string
     {
-        if ($options['hashid'] && !is_array($options['hashid'])) {
-            $name = $options['hashid'];
+        $name = $configuration->getName();
+
+        if (empty($name)) {
+            return null;
+        }
+
+        $hash = $request->attributes->get('_hash_' . $name);
+
+        if (!isset($hash)) {
+            $hash = $this->getHashFromAliases($hash, $request, $configuration);
         }
 
-        if ($request->attributes->has($name)) {
-            return (string) $request->attributes->get($name);
+        if (!isset($hash) && $this->autoConvert) {
+            $hash = (string) $request->attributes->get($name);
         }
```

Comment:
I think this fallback case should be placed before : 
```php
if (!isset($hash)) {
    $hash = $this->getHashFromAliases($hash, $request, $configuration);
}
```

This way we check in order if we find : 
* _hash_name
* name (if autoconvert)
* hashid/id

Reply:
I like your comment.

I will add this comment to simplify its understanding.

Reply:
Done